packages/modal/src/Modal.tsx

```diff
@@ -0,0 +1,281 @@
+`;
+
+const modalBodyStyle = css`
+  padding: 18px 0;
+`;
+
+const modalSizes: { readonly [K in ModalSize]: string } = {
+  xxsmall: css`
+    min-width: inherit;
+    max-width: inherit;
+    width: 366px;
+  `,
+
+  xsmall: css`
+    min-width: inherit;
+    max-width: inherit;
+    width: 418px;
+  `,
+
+  small: css`
+    min-width: inherit;
+    max-width: inherit;
+    width: 562px;
+  `,
+
+  normal: css`
+    min-width: inherit;
+    max-width: inherit;
+    width: 700px;
+  `,
+
+  large: css`
+    min-width: inherit;
+    max-width: inherit;
+    width: 800px;
+  `,
+
+  xlarge: css`
+    min-width: inherit;
+    max-width: inherit;
+    width: 1270px;
+  `,
+};
+
+const closeButton = css`
+  font-weight: 200;
+  color: #464c4f;
+  position: relative;
+  float: right;
+  top: -18px;
+  right: -10px;
+  margin-top: -2px;
+  cursor: pointer;
+`;
+
+const titleStyle = css`
+  margin: 0;
+`;
+
+const EscapeKey = 27;
+
+interface ModalProps {
+  /**
+   * Content that will appear inside of the Modal component.
+   */
+  children: React.ReactNode;
+
+  /**
+   * Determines the active state of the modal
+   *
+   * default: `false`
+   */
+  active: boolean;
+
+  /**
+   * Specifies that the Modal content will appear portaled to the end of the DOM,
+   * rather than in the React DOM tree.
+   *
+   * default: `true`
+   */
+  usePortal?: boolean;
+
+  /**
+   * Specifies the size of the Modal.
+   *
+   * default: `normal`
+   */
+  size?: ModalSize;
+
+  /**
+   * Specifies the title of the Modal.
+   *
+   */
+  title?: string;
+
+  /**
+   * Callback to change the active state of the Modal.
+   *
+   */
+  setActive?: (
+    active: boolean,
+  ) => void | React.Dispatch<SetStateAction<boolean>>;
+
+  /**
+   * Callback invoked when Modal closes.
+   *
+   */
+  onRequestClose?: () => void;
+
+  /**
+   * className applied to root overlay div.
+   *
+   */
+  className?: string;
+}
+
+/**
+ * # Modal
+ *
+ *  Modals place content on top of main window.
+ *
+```
+<Modal
+  active
+  usePortal
+  size="large"
+  setActive={setActive}
+  title="My Modal"
+  onRequestClose={() => console.log('Modal is closing now!')}
+  >  
+  Modal content!
+</Modal>
+```
+ * @param props.active Boolean to describe whether or not Modal is active.
+ * @param props.usePortal Boolean to describe if content should be portaled to end of DOM, or appear in React DOM tree.
+ * @param props.size String to determine size of Modal. ['xxsmall', 'xsmall', 'small', 'normal', 'large', 'xlarge']
+ * @param props.setActive Callback to change the active state of Modal.
+ * @param props.children Content to appear inside of Modal container.
+ * @param props.title Title for the Modal, will appear inside header tags.
+ * @param props.onRequestClose Callback invoked when Modal is closed.
+ * @param props.className className applied to overlay div.
+ *
+ */
+function Modal({
+  active = false,
+  usePortal = true,
+  size = ModalSize.Normal,
```

Comment:
Maybe `Default` rather than `Normal`

Reply:
Thought this was our convention from before. Will update now!